src/Microsoft.Dnx.Compilation.CSharp/CompilationContext.cs

```diff
@@ -52,6 +55,8 @@ public class CompilationContext
 
         public IList<ICompileModule> Modules { get; }
 
+        public ImmutableArray<DiagnosticAnalyzer> DiagnosticAnalyzers { get; set; }
```

Comment:
Not aware of performance impact, but I'll research. The reason why this method returns an `ImmutableArray` is because Roslyn requires this type to operate. Either it returns an `ImmutableArray` here or the consumer need to wrap the collection into `ImmutableArray`.